gui/middleware/form.py

```diff
@@ -0,0 +1,47 @@
+from django.core.exceptions import ValidationError as DjangoValidationError
+
+from freenasUI.middleware.client import client
+
+from middlewared.client.client import ValidationErrors as MiddlewareValidationErrors
+
+
+class MiddlewareModelForm:
+    key_prefix = NotImplemented
+    is_singletone = NotImplemented
+    middleware_plugin = NotImplemented
+
+    def clean(self):
+        try:
+            self.__update(dry_run=True)
```

Comment:
I am curious about the reason for the dry run?

Doesn't all the validation occur before? Why would you need two calls to middleware?

Reply:
Validation occurs when calling middleware `update` method from form's `save` method. Raising any validation errors there will lead to HTTP 500, so we should do validation earlier, in `clean` method as this is supposed to be with Django.

Reply:
The freeadmin code path handles ValidationErrors from middleware client, returning a valid 200 json response so I am wondering why that can be used or why its a bad idea to do so.

Reply:
I just did not knew about that code. I'll try to make use of it, thanks.